### DISCUSSION TITLE
Improve layout editor experience

### DISCUSSION THREAD

**Discussion by POeticPotatoes** [ppy/osu#21364] [Ideas]
(For osu! standard gamemode layout editing)

When the anchor points on an object are changed in the layout editor, the object might shift its position off the visible area which makes it only selectable through area selection (dragging to select a bunch of objects at a time). I'd suggest checking if the object will be out of the screen during repositioning and maintaining its current coordinates if so.

Also, you can only move an object by clicking on its actual pixels (instead of the selection box) which makes it really difficult to move things like healthbars (especially if you shrink them).

The UI rescales itself so that everything is visible during editing (which is cool) but it's very hard to work with the legacy objects  and estimate how they will show during actual gameplay (especially for skins with fullscreen designs).

I'm really loving lazer btw the new features are sick

**Comment by ghost** [comment]
you can also move objects via arrow keys in case that helps

**Reply by POeticPotatoes** [reply]
It does, but I think it would still be a much better experience if the draggable area was expanded to the entire object selection box.
